common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -1902,6 +1902,8 @@ bool ap_wlan_hal_dwpal::process_dwpal_event(char *buffer, int bufLen, const std:
         msg->params.vap_id       = beerocks::utils::get_ids_from_iface_string(VAP).vap_id;
         msg->params.mac          = beerocks::net::network_utils::mac_from_string(MACAddress);
         msg->params.capabilities = {};
+        std::copy_n(assoc_req, strnlen(assoc_req, ASSOCIATION_FRAME_SIZE) + 1,
+                    msg->params.association_frame);
```

Comment:
As I note elsewhere: you shouldn't simply copy, but convert to binary.

Reply:
I changed it but did not push it yet since I want to validate it works on testbed first (but I can't since the master branch using it now).

Currently, I'm facing an issue in which the test sometimes passes and sometimes not, without changing a thing (on a side branch).
I'll try to convert as you suggested in the ap_wlan_hal_dwpal.cpp (currently I'm converting it only before sending it as part of the client capability report message and maybe it's wrong..)